File: Year-2023/LeetCode-October-Challenge/NumberofGoodPairs.cpp
#include<bits/stdc++.h>
using namespace std;

class Solution {
public:
    int numIdenticalPairs(vector<int>& nums) {
        int res = 0;
        unordered_map<int, int> mp;
        for(auto &x: nums) mp[x]++;

        for(auto &x: mp){
            int n = x.second;
            res += ((n) * (n - 1))/2;
        }
        return res;
    }
};

int main(){
    vector<int> nums = {1,2,3,1,1,3};
    Solution *s = new Solution();
    cout<<s->numIdenticalPairs(nums)<<endl;
    return 0;
}

File: Year-2023/LeetCode-October-Challenge/NumberofGoodPairs.js
var numIdenticalPairs = function(nums) {
    let res = 0;
    let mp = {};
    for(let i of nums) mp[i] ? mp[i]++ : mp[i] = 1;

    for(let x in mp){
        let n = mp[x];
        res += ((n) * (n - 1))/2;
    }
    return res;
};

let nums = [1,2,3,1,1,3];
console.log(numIdenticalPairs(nums));